### ISSUE TITLE
Support GoToSocial (blocked on GoToSocial)

### ISSUE THREAD

**Issue by kvibber** [snarfed/bridgy-fed#1033]
Following up on #995: My GoToSocial server is unable to discover @bsky.brid.gy@bsky.brid.gy at all. It can retrieve the profile for my bridged Bluesky account @kelsonv.bsky.social@bsky.brid.gy, but the follow request doesn't seem to reach Bluesky.

Some possibly relevant logs from the GoToSocial side:
```
timestamp="07/05/2024 21:00:50.575" func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://bsky.brid.gy/.well-known/webfinger?resource=acct%3Absky.brid.gy%40bsky.brid.gy requestID=hhsnrnmf040004rs2zsg pubKeyID=https://notes.kvibber.com/users/kelson/main-key msg="performing request"
timestamp="07/05/2024 21:00:51.054" func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://bsky.brid.gy/bsky.brid.gy requestID=hhsnrnmf040004rs2zsg pubKeyID=https://notes.kvibber.com/users/kelson/main-key msg="performing request"
timestamp="07/05/2024 21:00:51.472" func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://fed.brid.gy/static/bridgy_logo_square.jpg requestID=hhsnrnmf040004rs2zsg pubKeyID=https://notes.kvibber.com/users/kelson/main-key msg="performing request"
timestamp="07/05/2024 21:00:51.609" func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://fed.brid.gy/static/bridgy_logo.jpg requestID=hhsnrnmf040004rs2zsg pubKeyID=https://notes.kvibber.com/users/kelson/main-key msg="performing request"
timestamp="07/05/2024 21:01:43.903" func=server.glob..func1.Logger.func13.1 level=INFO latency="11.004341ms" userAgent="Bridgy Fed (https://fed.brid.gy/)" method=GET statusCode=404 path=/@kelsonvibber.com clientIP=172.18.0.1 pubKeyID=https://fed.brid.gy/fed.brid.gy#key errors="Error #01: account with username kelsonvibber.com not found in the db\n" requestID=th1ntnmf04001pbpexsg msg="Not Found: wrote 1.21kiB"
```
The last line is particularly weird, since that's neither the username nor the domain I'm using on this server.

**Comment by snarfed** [comment]
Hmm! We had some problems with other servers discovering `@bsky.brid.gy@bsky.brid.gy` too, #995, but I'd hoped we'd resolved those.

Thanks for the logs, they're useful. Looks like your server is fetching and parsing our Webfinger and then fetching the AP actor https://bsky.brid.gy/bsky.brid.gy, which is good. I see in our logs that we returned 200 to that request with valid a AS2 JSON actor. We serve it with `Content-Type: application/ld+json; profile="https://www.w3.org/ns/activitystreams"`, does GoToSocial maybe only handle `application/activity+json`?

(I'm pretty sure the last line is unrelated, I see why BF fetches that, it's a separate thing, and also it was >50s later here.)

Otherwise, you can see the https://bsky.brid.gy/bsky.brid.gy AS2 JSON in a browser, feel free to send it to the GoToSocial maintainers if they want to look!

**Comment by WinsDominoes** [comment]
That's interesting, seems to work for me however. 

![Screenshot_20240509_044239_Moshidon.jpg](https://github.com/snarfed/bridgy-fed/assets/49825651/7061cc84-a790-4342-9464-50601a303ed3)

**Comment by kvibber** [comment]
I turned on TRACE logging to search for the opt-in account. GoToSocial is getting the response and parsing it successfully. But when it tries to add it to the local database, it's running into a constraint violation on the url field and rolling back the insert.

@fed.brid.gy@fed.brid.gy is in my db with https://fed.brid.gy/fed.brid.gy as the URI and https://fed.brid.gy/ as the URL.

It's trying to add @bsky.brid.gy@bsky.brid.gy with https://bsky.brid.gy/bsky.brid.gy as the URI...and https://fed.brid.gy/ again as the URL.

It looks like it's using the `id` and `url` fields from the JSON object, since my own bridged account is stored with URI https://bsky.brid.gy/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo and URL https://bsky.brid.gy/r/https://bsky.app/profile/kelsonv.bsky.social

But that would explain why @WinsDominoes can find it - if their DB doesn't already have a reference to @fed.brid.gy@fed.brid.gy, it'll be able to insert the record for @bsky.brid.gy@bsky.brid.gy

I don't know the spec very well, so I'm not sure if the `url` field *should* be expected to be unique per actor or not, especially since I see @bsky.brid.gy has multiple values for the field.

So I don't know if the problem is that the field needs to be unique in the actual JSON, or that GoToSocial needs to change the data constraints.

The GTS team is on vacation this month, but if it looks like the change needs to be on their end, I can try the schema change on my test server, see if it helps, and file an issue when they get back and reopen the repo.

I'll try to get some debug info on the follow requests tomorrow evening if I have time.

**Comment by snarfed** [comment]
Great sleuthing! Sounds like you found the problem.

AP `id`s do need to be unique across objects - it's a SHOULD in AS2 and a MUST in AP - but afaik nothing else does. I can think of plenty of use cases for multiple actors having the same `url` - alt accounts, employees at the same company, bot users for the same app like here, etc. Seems like a schema bug on GTS's part.

**Comment by kvibber** [comment]
Aha! The extra constraint on URL is a known issue in GTS:
https://github.com/superseriousbusiness/gotosocial/issues/1820
https://github.com/superseriousbusiness/gotosocial/issues/2082

One of those cites a couple of places where they do lookups assuming it's unique, so those need to be fixed in order to safely change the schema.

So the *discovery* part is definitely covered by that issue!

That leaves the Follow part, which I still need to collect logs for.

**Comment by kvibber** [comment]
OK, I *think* the key lines related to trying to follow my bsky account from gts are these:

```
timestamp="10/05/2024 03:56:26.136" func=federatingdb.(*federatingDB).Accept level=DEBUG accept="{\"@context\":\"https://www.w3.org/ns/activitystreams\",\"actor\":\"https://bsky.brid.gy/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo\",\"id\":\"https://bsky.brid.gy/convert/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo/followers#accept-https://gts.keysmash.xyz/users/addhawk/follow/012FE5QQ18RX7FAQZDQZW40MNT\",\"object\":{\"actor\":\"https://gts.keysmash.xyz/users/addhawk\",\"id\":\"https://gts.keysmash.xyz/users/addhawk/follow/012FE5QQ18RX7FAQZDQZW40MNT\",\"object\":\"did:plc:atsrwbmgqyeko5cjtzt5cymo\",\"to\":[\"did:plc:atsrwbmgqyeko5cjtzt5cymo\",\"https://www.w3.org/ns/activitystreams#Public\"],\"type\":\"Follow\",\"url\":\"https://gts.keysmash.xyz/users/addhawk#followed-did:plc:atsrwbmgqyeko5cjtzt5cymo\"},\"to\":\"https://www.w3.org/ns/activitystreams#Public\",\"type\":\"Accept\"}" requestID=j0haar4f04001qe3avqg msg="entering Accept"
timestamp="10/05/2024 03:56:26.137" func=server.glob..func1.Logger.func13.1 level=ERROR latency="9.433743ms" userAgent="Bridgy Fed (https://fed.brid.gy/)" method=POST statusCode=500 path=/users/addhawk/inbox clientIP=107.178.200.209 pubKeyID=https://bsky.brid.gy/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo#key errors="Error #01: PostInboxScheme: error calling sideEffectActor.PostInbox: ACCEPT: error converting asfollow to gtsfollow: getASObjectAccount: error getting object account from database: sql: no rows in result set\n" requestID=j0haar4f04001qe3avqg msg="Internal Server Error: wrote 54B"

timestamp="10/05/2024 03:56:26.192" func=httpclient.(*Client).DoSigned level=ERROR method=POST url=https://bsky.brid.gy/ap/sharedInbox requestID=tg5aar4f040010wabgcg pubKeyID=https://gts.keysmash.xyz/users/addhawk/main-key msg="backing off for 4s after http request error: http response: 502 Bad Gateway"

timestamp="10/05/2024 03:56:30.299" func=workers.(*clientAPI).CreateFollowReq level=ERROR requestID=tg5aar4f040010wabgcg msg="error federating follow request: Follow: error sending activity *typefollow.ActivityStreamsFollow via outbox https://gts.keysmash.xyz/users/addhawk/outbox: func1: error delivering to https://bsky.brid.gy/ap/sharedInbox: deliver: POST request to https://bsky.brid.gy/ap/sharedInbox failed: status=\"204 No Content\" body=\"<empty>\""
```

I think GTS not being able to locate the account in the DB (except that this time I confirmed that the account *is* in the DB, with the correct URI) is likely a bug in GTS. I've got most of a month to look at that part again if I want to before submitting a bug report to them.

I'm not sure about the two attempts to contact the bridge's sharedInbox, though.

**Comment by snarfed** [comment]
Aha, useful! So in the process of handling the first shared inbox request, we synchronously POSTed our own `Follow` in response, below - of the bot user following you - to https://gts.keysmash.xyz/users/addhawk/inbox, and your instance responded with HTTP 500 `{"error":"Internal Server Error"}`. That may have the same root db cause.

```json
{
  "id": "https://bsky.brid.gy/convert/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo/followers#accept-https://gts.keysmash.xyz/users/addhawk/follow/015VX2246S40WTNVYE0T9E0WAW",
  "actor": "https://bsky.brid.gy/ap/did:plc:atsrwbmgqyeko5cjtzt5cymo",
  "@context": "https://www.w3.org/ns/activitystreams",
  "type": "Accept",
  "object": {
    "actor": "https://gts.keysmash.xyz/users/addhawk",
    "id": "https://gts.keysmash.xyz/users/addhawk/follow/015VX2246S40WTNVYE0T9E0WAW",
    "url": "https://gts.keysmash.xyz/users/addhawk#followed-did:plc:atsrwbmgqyeko5cjtzt5cymo",
    "type": "Follow",
    "object": "did:plc:atsrwbmgqyeko5cjtzt5cymo",
    "to": [
      "did:plc:atsrwbmgqyeko5cjtzt5cymo",
      "https://www.w3.org/ns/activitystreams#Public"
    ]
  },
  "to": [
    "https://www.w3.org/ns/activitystreams#Public"
  ]
}
```

Granted, we should send that `Follow` async, and not fail your incoming follow due to our own follow's failure. The second shared inbox delivery also should retry instead of skipping it and returning 204. I'll file BF issues for those.

**Comment by snarfed** [comment]
OK! Fixed #1041, which is the main reason BF was getting stuck on the 500 from sending GTS a `Follow` while processing its incoming `Follow`. That should hopefully unblock the BF side, so once GTS can handle multiple actors with the same `url`, you might be able to follow the bot user.

Let me know how it goes, happy to help more!

**Comment by snarfed** [comment]
Reopening to track our GoToSocial interop. Current status here is that GoToSocial (0.16.0, latest released version) currently 500s when we fetch it with an HTTP signature with `keyId=https://fed.brid.gy/fed.brid.gy#key`: https://github.com/superseriousbusiness/gotosocial/issues/3381

**Comment by qazmlp** [comment]
According to https://github.com/superseriousbusiness/gotosocial/blob/c023bd30f381d6532578ee0bc40d12704c2b97a6/internal/typeutils/astointernal.go#L173, a likely workaround would be to swap the order of the `"url"` array that's returned on `@bsky.brid.gy@bsky.brid.gy`.

I'm not sure how quickly that'd fix it on servers that already have this account in their database, but I suspect it'd bypass the problem on ones that so far have only `@fed.brid.gy@fed.brid.gy` pretty much instantly.

**Comment by Tamschi** [comment]
I'll remove the `blocked` label since we _could_ easily work around the current issue at least.

**Comment by eeeeeta** [comment]
While GoToSocial can now fetch the `bsky.brid.gy` user, it's still not possible to follow it. Logs from GtS below; I suspect the issue is the bridge doing authenticated fetch with the `fed.brid.gy` key, perhaps?

```
Nov 15 20:05:37 shenfield-nix gotosocial[276680]: func=server.init.func1.Logger.13.1 level=ERROR latency=1.066605641s userAgent="Bridgy Fed (https://fed.brid.gy/)" method=GET statusCode=500 path=/users/eta pubKeyID=https://fed.brid.gy/fed.brid.gy#key errors="Error #01: AuthenticateFederatedRequest: error dereferencing account https://fed.brid.gy/fed.brid.gy: enrichAccountSafely: error getting account https://fed.brid.gy/fed.brid.gy from database after race: sql: no rows in result set\n" requestID=m9e6wcck04001qky4bw0 msg="Internal Server Error: wrote 1.00kiB"
```

**Comment by Tamschi** [comment]
That's the old bug, yes.

Can you cause the instance to refetch/update `@bsky.brid.gy@bsky.brid.gy`?
It should have the `url` `https://bsky.brid.gy/` afterwards, which frees up `https://fed.brid.gy/` for `@fed.brid.gy@fed.brid.gy`.

**Comment by eeeeeta** [comment]
As I understand it, the problem is not `https://fed.brid.gy` being taken, it's that the bridge is making requests with a key `https://fed.brid.gy/fed.brid.gy#key`, which causes GtS to try and fail to look that up.

I bodged it like this, and got the bridge to mostly work:
```
sqlite> UPDATE accounts SET url = 'https://fed.brid.gy/fed.brid.gy' WHERE url = 'https://fed.brid.gy';
```

Weirdly I now have two of them:

<details><summary>sqlite database output</summary>
<pre>
sqlite> SELECT * FROM accounts WHERE url = 'https://fed.brid.gy/';
                        id = 01JCRQHFA6ZFSA5DDXH4Z4FA3R
                created_at = 2024-11-15 20:16:56.683142+00:00
                updated_at = 2024-11-15 20:16:56.683142+00:00
                fetched_at = 2024-11-15 20:16:56.683142+00:00
                  username = fed.brid.gy
                    domain = fed.brid.gy
avatar_media_attachment_id = 01JCRQHFBBXSYCWNFSVGRT5VX9
         avatar_remote_url = https://fed.brid.gy/static/bridgy_logo_square.jpg
header_media_attachment_id = 01JCRQHFJ4CKKJC00ZG2WYHPEB
         header_remote_url = https://fed.brid.gy/static/bridgy_fed_banner.png
              display_name = Bridgy Fed
                    emojis = null
                    fields = null
                fields_raw = null
                      note = <a href="https://fed.brid.gy/" rel="nofollow noreferrer noopener" target="_blank">Bridgy Fed</a> is a bridge between decentralized social networks like the <a href="https://indieweb.org/" rel="nofollow noreferrer noopener" target="_blank">IndieWeb</a>, <a href="https://en.wikipedia.org/wiki/Fediverse" rel="nofollow noreferrer noopener" target="_blank">fediverse</a>, <a href="https://bsky.social/" rel="nofollow noreferrer noopener" target="_blank">Bluesky</a>, and others. <a href="https://fed.brid.gy/docs" rel="nofollow noreferrer noopener" target="_blank">More info here.</a>
                  note_raw = 
                  memorial = 0
                       bot = 1
                    locked = 1
              discoverable = 0
                       uri = https://fed.brid.gy/fed.brid.gy
                       url = https://fed.brid.gy/
                 inbox_uri = https://fed.brid.gy/fed.brid.gy/inbox
          shared_inbox_uri = https://web.brid.gy/ap/sharedInbox
                outbox_uri = https://fed.brid.gy/fed.brid.gy/outbox
             following_uri = https://fed.brid.gy/fed.brid.gy/following
             followers_uri = https://fed.brid.gy/fed.brid.gy/followers
   featured_collection_uri = 
                actor_type = Application
               private_key = 
                public_key = {"N":19086279233169020349192059689095836109398073954970021960930094516078417621184632944226647104938930843509708224671637763291274700600219006379085971423674967673991604978212422869676544214149539308800434426486918105134222702089925275950630755331516635692291258590592206061429747183887180147349018066861611836960803945376945234918904904061857800412842583032159150976623242576886922171076759834609466509564094008513359614213287367050678004119092016769549824252851267796191985400591509438743289115102691701592785990614370076528839419173141277249730942474779752241907587872581899336638856293133797513866512743681525158220667,"E":65537}
            public_key_uri = https://fed.brid.gy/fed.brid.gy#key
             sensitized_at = 
               silenced_at = 
              suspended_at = 
         suspension_origin = 
     public_key_expires_at = 
              moved_to_uri = 
        also_known_as_uris = ["https://fed.brid.gy/"]
                   move_id = 
sqlite> SELECT * FROM accounts WHERE url = 'https://fed.brid.gy/fed.brid.gy';
                        id = 01JCRPWDQ3KZTD5T6EB5XSCXGB
                created_at = 2024-11-15 20:05:27.166682+00:00
                updated_at = 2024-11-15 20:05:27.166682+00:00
                fetched_at = 2024-11-15 20:05:27.166682+00:00
                  username = bsky.brid.gy
                    domain = bsky.brid.gy
avatar_media_attachment_id = 01JCRPWDZYZ9D4RW3VMMMCZ12W
         avatar_remote_url = https://fed.brid.gy/static/bridgy_logo_square.jpg
header_media_attachment_id = 01JCRPWE86F61GEDB70WC7373F
         header_remote_url = https://fed.brid.gy/static/bridgy_fed_banner.png
              display_name = Bridgy Fed for Bluesky
                    emojis = null
                    fields = null
                fields_raw = null
                      note = <p><a href="https://fed.brid.gy/" rel="nofollow noreferrer noopener" target="_blank">Bridgy Fed</a> bot user for <a href="https://bsky.social/" rel="nofollow noreferrer noopener" target="_blank">Bluesky</a>. To bridge your fediverse account to Bluesky, follow this account. <a href="https://fed.brid.gy/docs" rel="nofollow noreferrer noopener" target="_blank">More info here.</a><p>After you follow this account, it will follow you back. Accept its follow to make sure your fediverse posts get sent to the bridge and make it into Bluesky.<p>To ask a Bluesky user to bridge their account, DM their handle (eg snarfed.bsky.social) to this account.</p>
                  note_raw = 
                  memorial = 0
                       bot = 1
                    locked = 1
              discoverable = 0
                       uri = https://bsky.brid.gy/bsky.brid.gy
                       url = https://fed.brid.gy/fed.brid.gy
                 inbox_uri = https://bsky.brid.gy/bsky.brid.gy/inbox
          shared_inbox_uri = https://web.brid.gy/ap/sharedInbox
                outbox_uri = https://bsky.brid.gy/bsky.brid.gy/outbox
             following_uri = https://bsky.brid.gy/bsky.brid.gy/following
             followers_uri = https://bsky.brid.gy/bsky.brid.gy/followers
   featured_collection_uri = 
                actor_type = Application
               private_key = 
                public_key = {"N":19546921171224486569000798086203896617302539369877918710669989290972022979993009010499098555981408896793967039487101112510275121067295063192196768222135561927639888231387424389319813594475170504181851169982631189308194193738818194065752156655674130636086683242629765582749720265905967971750882772670767637792509118289703188592798463421214513963016263833383806211269121778941509475696888099567839560949356225915939071512882905315839289265148880265309249111802380338704002053827879714738318983575258315853505172543645187370416083379460249152142666994042765068030366157600543639801197196845250195266628473303886310517077,"E":65537}
            public_key_uri = https://bsky.brid.gy/bsky.brid.gy#key
             sensitized_at = 
               silenced_at = 
              suspended_at = 
         suspension_origin = 
     public_key_expires_at = 
              moved_to_uri = 
        also_known_as_uris = ["https://bsky.brid.gy/","https://fed.brid.gy/"]
                   move_id = 
</pre>
</details>

Anyway, this moved me on to the next series of errors: following bridged bsky accounts fails with the following.

```
Nov 15 20:34:54 shenfield-nix gotosocial[276680]: func=server.init.func1.Logger.13.1 level=ERROR latency=2.34033ms userAgent="Bridgy Fed (https://fed.brid.gy/)" method=POST statusCode=500 path=/users/eta/inbox clientIP=34.31.157.142 pubKeyID=https://bsky.brid.gy/ap/did:plc:inze6wrmsm7pjl7yta3oig77#key errors="Error #01: acceptFollowType: error converting Follow to *gtsmodel.Follow: getASObjectAccount: error getting object account from database: sql: no rows in result set\n" requestID=t4r8jcck040019748dg0 msg="Internal Server Error: wrote 54B"
```

**Comment by Tamschi** [comment]
> As I understand it, the problem is not `https://fed.brid.gy` being taken, it's that the bridge is making requests with a key `https://fed.brid.gy/fed.brid.gy#key`, which causes GtS to try and fail to look that up.

Yes, which fails only because it can't put the signing user into its database when the `url` collides.

> […]
> 
> Anyway, this moved me on to the next series of errors: following bridged bsky accounts fails with the following.
> 
> ```
> Nov 15 20:34:54 shenfield-nix gotosocial[276680]: func=server.init.func1.Logger.13.1 level=ERROR latency=2.34033ms userAgent="Bridgy Fed (https://fed.brid.gy/)" method=POST statusCode=500 path=/users/eta/inbox clientIP=34.31.157.142 pubKeyID=https://bsky.brid.gy/ap/did:plc:inze6wrmsm7pjl7yta3oig77#key errors="Error #01: acceptFollowType: error converting Follow to *gtsmodel.Follow: getASObjectAccount: error getting object account from database: sql: no rows in result set\n" requestID=t4r8jcck040019748dg0 msg="Internal Server Error: wrote 54B"
> ```

Progress 🙂
Maybe.

The error is thrown here: https://github.com/superseriousbusiness/gotosocial/blob/e892bc375d733eb73488958b26c73fba7e83f7e8/internal/typeutils/astointernal.go#L859

I'll look through a bit to see how that's hit precisely, if I can see it by just reading the code.

**Comment by eeeeeta** [comment]
I enabled trace logging in GoToSocial and retried a follow. Looks like it's upset with this:

```json
{
    "@context": "https://www.w3.org/ns/activitystreams",
    "actor": "https://bsky.brid.gy/ap/did:plc:inze6wrmsm7pjl7yta3oig77",
    "id": "https://bsky.brid.gy/convert/ap/did:plc:inze6wrmsm7pjl7yta3oig77/followers#accept-https://gotosocial.i.eta.st/users/eta/follow/01JYZNV5JJ9GP3QQTDJJG4HP5X",
    "object": {
        "actor": "https://gotosocial.i.eta.st/users/eta",
        "id": "https://gotosocial.i.eta.st/users/eta/follow/01JYZNV5JJ9GP3QQTDJJG4HP5X",
        "object": "did:plc:inze6wrmsm7pjl7yta3oig77",
        "to": "https://bsky.brid.gy/ap/did:plc:inze6wrmsm7pjl7yta3oig77",
        "type": "Follow",
        "url": "https://gotosocial.i.eta.st/users/eta#followed-did:plc:inze6wrmsm7pjl7yta3oig77"
    },
    "to": "https://www.w3.org/ns/activitystreams#Public",
    "type": "Accept"
}
```

I would hazard a guess that it's looking at the `object` property, and not the `to`  (this seems to point in that direction: https://github.com/superseriousbusiness/gotosocial/blob/main/internal/typeutils/astointernal.go#L465). Putting a bsky DID in there is certainly interesting -- I'm surprised that doesn't cause more problems!

**Comment by Tamschi** [comment]
Yeah, that is definitely not compliant on our end. I think other software probably just looks at the `Follow`'s `id` to figure out the association.

I'll file a separate issue.

**Comment by eeeeeta** [comment]
@snarfed Yes, I can follow bridged bluesky users now, thank you!

**Comment by snarfed** [comment]
Exciting! Thanks for confirming!

I've lost the plot here overall a bit...sounds like GTS still needs to fix https://github.com/snarfed/bridgy-fed/issues/1033#issuecomment-2479889372 ? And nothing else outstanding on our side?

**Comment by Tamschi** [comment]
> I've lost the plot here overall a bit...sounds like GTS still needs to fix [#1033 (comment)](https://github.com/snarfed/bridgy-fed/issues/1033#issuecomment-2479889372) ? And nothing else outstanding on our side?

For instances where the `@bsky.brid.gy@bsky.brid.gy` actor is loaded but stale from before the workaround in #1445, yes.
Since we have a workaround now, deleting `@bsky.brid.gy@bsky.brid.gy` from the database would fix it, though, without the proper `"url"` non-uniqueness fix on their end.

So _in theory_ bridging could work already, as long as both `@bsky.brid.gy@bsky.brid.gy` and `@fed.brid.gy@fed.brid.gy` are accessible on a server. We just need someone to test it.

**Comment by kvibber** [comment]
It does!

I deleted `@bsky.brid.gy@bsky.brid.gy` from my GoToSocial `accounts` table, tried to follow it again, and *this* time it worked, and I was able to approve the follow-back from the bridge.

Over on Bluesky, I was able to find and follow my GoToSocial profile, and the next post I made on GTS did in fact get bridged!

https://bsky.app/profile/kelson.notes.kvibber.com.ap.brid.gy/post/3lc7cqax2sh42

Now to test the other direction....

**Comment by snarfed** [comment]
Great news! Thanks all! Now to get GTS to fix the actual bug so that admins don't all have to do that manually.

**Comment by kvibber** [comment]
And I can follow a Bluesky account from GoToSocial now! I had to withdraw the request, follow, unfollow, and follow again, but hopefully that's only because I'd already tried to follow that account in the past.

**Comment by meljoann** [comment]
Hi, I'm trying to bridge as well, but it seems stuck in "Request follow". For the below workaround, did you suspend it from the admin panel or do something with command line or Postgres?

> I deleted `@bsky.brid.gy@bsky.brid.gy` from my GoToSocial `accounts` table, tried to follow it again, and _this_ time it worked, and I was able to approve the follow-back from the bridge.

**Comment by kvibber** [comment]
I deleted it in the database (Sqlite in my case). I think it was just that table.

**Comment by snarfed** [comment]
Good news, GtS removed the url constraint a bit ago, https://codeberg.org/superseriousbusiness/gotosocial/pulls/3928 , and released it in 0.19.0. I think that was the last open issue here, Tentatively closing.
